app/src/main/java/com/example/taskyy/domain/usecases/LoginUseCase.kt

```diff
@@ -1,10 +1,42 @@
 package com.example.taskyy.domain.usecases
 
 import android.util.Patterns
+import com.example.taskyy.data.remote.LoginUserResponse
+import com.example.taskyy.domain.objects.Login
+import com.example.taskyy.domain.repository.AuthRepository
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import javax.inject.Inject
```

Comment:
Here you can see how much your use case is coupled to other libraries, which is not allowed as described in my previous comment.
Instead, the repository should execute the Retrofit call and do the response handling. Then all the info the use case gets is whether logging in was successful or not (with potential data or errors)